### ISSUE TITLE
[JENKINS-2997] multiple parallel parameterized builds seem to happen at once

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#14142]
<p>Currently Hudson does not allow multiple builds in parallel for the same <br/>
project. However when a project is parameterized, this <em>IS</em> possible.</p>

<p>This is very dangerous, since it's perfectly possible that two builds will be <br/>
started on the same workspace. Imagine your workspace being updated by one build <br/>
while the other one is still doing the build. You might not even notice <br/>
something wrong has happened but your build results would be completely bogus.</p>

<p>The current workaround is to make sure you do only one parallel build per node.</p>

<p>The immediate solution is to disable parallel builds in the core, but this <br/>
requires implementing equals() and hashcode() differently on <em>ALL</em> Queue.Task <br/>
implementers.</p>

<p>The final solution is to implement decent locking, and fix all other places in <br/>
Hudson that depend on only having one build running.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=huybrechts">huybrechts</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997" target="_blank">multiple parallel parameterized builds seem to happen at once</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: core
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2014-10-07T19:02:01+00:00
<li><b>votes</b>: 3
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Currently Hudson does not allow multiple builds in parallel for the same 
project. However when a project is parameterized, this <em>IS</em> possible.</p>

<p>This is very dangerous, since it's perfectly possible that two builds will be 
started on the same workspace. Imagine your workspace being updated by one build 
while the other one is still doing the build. You might not even notice 
something wrong has happened but your build results would be completely bogus.</p>

<p>The current workaround is to make sure you do only one parallel build per node.</p>

<p>The immediate solution is to disable parallel builds in the core, but this 
requires implementing equals() and hashcode() differently on <em>ALL</em> Queue.Task 
implementers.</p>

<p>The final solution is to implement decent locking, and fix all other places in 
Hudson that depend on only having one build running.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>
<details><summary><i>4 attachments</i></summary>

- [corrupted_build_history.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18492/corrupted_build_history.png)
  > ![corrupted_build_history.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18492/corrupted_build_history.png)
- [corrupted_build_history_2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18493/corrupted_build_history_2.png)
  > ![corrupted_build_history_2.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18493/corrupted_build_history_2.png)
- [queue-actions-2.diff](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18491/queue-actions-2.diff)
- [queue-actions.diff](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/18490/queue-actions.diff)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128662&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128662">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>working on it</p>
</pre>
</details>

</ul>
</details>

<p>working on it</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128663&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128663">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created an attachment (id=552)
proposed patch</p>
</pre>
</details>

</ul>
</details>

<p>Created an attachment (id=552)<br/>
proposed patch</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128664&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128664">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>proposed patch passes Actions when queuing a build</p>

<p>TODO:</p>
<ul class="alternate" type="square">
	<li>verify queue.xml serialization (backward compatible ?)</li>
</ul>


<p>TODO (optional):</p>
<ul class="alternate" type="square">
	<li>implement an optional interface BuildWrappingAction that can influence the build</li>
	<li>implement parameters on top of this</li>
	<li>implement UI when scheduling a build</li>
	<li>implement parameter asking on top of this</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>proposed patch passes Actions when queuing a build</p>

<p>TODO:</p>
<ul class="alternate" type="square">
	<li>verify queue.xml serialization (backward compatible ?)</li>
</ul>


<p>TODO (optional):</p>
<ul class="alternate" type="square">
	<li>implement an optional interface BuildWrappingAction that can influence the build</li>
	<li>implement parameters on top of this</li>
	<li>implement UI when scheduling a build</li>
	<li>implement parameter asking on top of this</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128665&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128665">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>one comment on the patch: I see one spot in AbstractProject diffs where
quietPeriod was replaced with zero.</p>

<p>one question: before this change you could go schedule several builds with
different parameters.. the dangerous part was these builds running in parallel,
but I guess it was good they could all be in the queue.  with this patch what
happens if you submit a parameterized build when one for the same project is
already in the queue (even with different parameters)?</p>
</pre>
</details>

</ul>
</details>

<p>one comment on the patch: I see one spot in AbstractProject diffs where<br/>
quietPeriod was replaced with zero.</p>

<p>one question: before this change you could go schedule several builds with<br/>
different parameters.. the dangerous part was these builds running in parallel,<br/>
but I guess it was good they could all be in the queue.  with this patch what<br/>
happens if you submit a parameterized build when one for the same project is<br/>
already in the queue (even with different parameters)?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128666&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128666">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created an attachment (id=554)
This version allows multiple scheduled builds for parameterized projects, but no parallel building. Still WIP.</p>
</pre>
</details>

</ul>
</details>

<p>Created an attachment (id=554)<br/>
This version allows multiple scheduled builds for parameterized projects, but no parallel building. Still WIP.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128667&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128667">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>patch is getting long.. I didn't make it thru all the Queue.java stuff yet, but
did notice: diff in MatrixConfiguration doesn't use the Cause parameter anymore</p>
</pre>
</details>

</ul>
</details>

<p>patch is getting long.. I didn't make it thru all the Queue.java stuff yet, but<br/>
did notice: diff in MatrixConfiguration doesn't use the Cause parameter anymore</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128668&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128668">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Looks like ParametersAction always allows the same job to be added in the queue.
 In previous(current) code it was using equality of the Task objects to avoid
duplicates, so a job submitted with the same set of parameters would not get
scheduled again.
Seems like this aspect is somewhat important, as jobs with parameters may still
be invoked via SCM or Timer with the default parameters.. if a build is blocked
(waiting on executor, etc) and it keeps polling and triggering, we could end up
with tons of scheduled jobs with the same default parameters.  What do you think?</p>
</pre>
</details>

</ul>
</details>

<p>Looks like ParametersAction always allows the same job to be added in the queue.<br/>
 In previous(current) code it was using equality of the Task objects to avoid<br/>
duplicates, so a job submitted with the same set of parameters would not get<br/>
scheduled again.<br/>
Seems like this aspect is somewhat important, as jobs with parameters may still<br/>
be invoked via SCM or Timer with the default parameters.. if a build is blocked<br/>
(waiting on executor, etc) and it keeps polling and triggering, we could end up<br/>
with tons of scheduled jobs with the same default parameters.  What do you think?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128669&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128669">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>r15291 | huybrechts | 2009-02-13 09:59:04 -0700 (Fri, 13 Feb 2009) | 1 line
Changed paths:
   M /trunk/hudson/main/core/src/main/java/hudson/matrix/MatrixConfiguration.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/Executor.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/FileParameterValue.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParameterValue.java
   D
/trunk/hudson/main/core/src/main/java/hudson/model/ParameterizedProjectTask.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParametersAction.java
   M
/trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/Queue.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/StringParameterValue.java
   D /trunk/hudson/main/core/src/main/java/hudson/util/QueueTaskFilter.java
   M /trunk/hudson/main/core/src/main/java/hudson/widgets/BuildHistoryWidget.java
   M
/trunk/hudson/main/core/src/main/resources/hudson/widgets/BuildHistoryWidget/entries.jelly</p>

<p><span class="error">&#91;FIXED JENKINS-2997&#93;</span> introducing QueueAction to pass actions from scheduling to
the build it self</p>
</pre>
</details>

</ul>
</details>

<p>r15291 | huybrechts | 2009-02-13 09:59:04 -0700 (Fri, 13 Feb 2009) | 1 line<br/>
Changed paths:<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/matrix/MatrixConfiguration.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/Executor.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/FileParameterValue.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParameterValue.java<br/>
   D<br/>
/trunk/hudson/main/core/src/main/java/hudson/model/ParameterizedProjectTask.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParametersAction.java<br/>
   M<br/>
/trunk/hudson/main/core/src/main/java/hudson/model/ParametersDefinitionProperty.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/Queue.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/StringParameterValue.java<br/>
   D /trunk/hudson/main/core/src/main/java/hudson/util/QueueTaskFilter.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/widgets/BuildHistoryWidget.java<br/>
   M<br/>
/trunk/hudson/main/core/src/main/resources/hudson/widgets/BuildHistoryWidget/entries.jelly</p>

<p><span class="error">&#91;FIXED JENKINS-2997&#93;</span> introducing QueueAction to pass actions from scheduling to<br/>
the build it self</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128670&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128670">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>On current svn, I just tried this:
1) create job with 1 string parameter, job just does "sleep 90"
2) start job (starts immediately)
3) start job again (now one in queue)
4) start job again with different parameter value (now two in queue)</p>

<p>when the first build completed, both jobs in the queue started simultaneously.</p>
</pre>
</details>

</ul>
</details>

<p>On current svn, I just tried this:<br/>
1) create job with 1 string parameter, job just does "sleep 90"<br/>
2) start job (starts immediately)<br/>
3) start job again (now one in queue)<br/>
4) start job again with different parameter value (now two in queue)</p>

<p>when the first build completed, both jobs in the queue started simultaneously.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128671&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128671">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>also, the "others.isEmpty" case in ParametersAction.shouldSchedule looks wrong..
it returns parameters.isEmpty().. should it be !parameters.isEmpty() ?</p>
</pre>
</details>

</ul>
</details>

<p>also, the "others.isEmpty" case in ParametersAction.shouldSchedule looks wrong..<br/>
it returns parameters.isEmpty().. should it be !parameters.isEmpty() ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128672&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128672">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul class="alternate" type="square">
	<li>the isEmpty() is corrected</li>
</ul>


<ul class="alternate" type="square">
	<li>the parallel building you see is an illusion... There is a gap (you might call
it a bug) in the Queue implementation. It checks for resource locks before 
handing over an item in the queue to an executor, but doesn actually reserve the 
resources. Then before the executor starts building, the resources are actually 
reserverd. Therefore it is possible that more than one build gets passed the 
first check. They will then all be assigned an executor, but still only one can 
build at the same time. When multiple builds are scheduled, this is very likely 
to happen. The bad effect here is that an executor is unnecessarily occupied, 
but there is still no parallel building.</li>
</ul>


<p>But... when two builds on the same project start in the same second, they will 
have the same build id (not to be confused with the build number), and thus 
share the same build directory ! Really bad... Because of this, the two builds 
that look like they are building in parallel in the previous example will even 
share the same log file. And if you look at the console output of both these 
builds, it looks like both are building.</p>
</pre>
</details>

</ul>
</details>

<ul class="alternate" type="square">
	<li>the isEmpty() is corrected</li>
</ul>


<ul class="alternate" type="square">
	<li>the parallel building you see is an illusion... There is a gap (you might call<br/>
it a bug) in the Queue implementation. It checks for resource locks before <br/>
handing over an item in the queue to an executor, but doesn actually reserve the <br/>
resources. Then before the executor starts building, the resources are actually <br/>
reserverd. Therefore it is possible that more than one build gets passed the <br/>
first check. They will then all be assigned an executor, but still only one can <br/>
build at the same time. When multiple builds are scheduled, this is very likely <br/>
to happen. The bad effect here is that an executor is unnecessarily occupied, <br/>
but there is still no parallel building.</li>
</ul>


<p>But... when two builds on the same project start in the same second, they will <br/>
have the same build id (not to be confused with the build number), and thus <br/>
share the same build directory ! Really bad... Because of this, the two builds <br/>
that look like they are building in parallel in the previous example will even <br/>
share the same log file. And if you look at the console output of both these <br/>
builds, it looks like both are building.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128673&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128673">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>r15338 | huybrechts | 2009-02-14 04:36:57 -0800 (Sat, 14 Feb 2009) | 3 lines
Changed paths:
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParametersAction.java</p>

<p><span class="error">&#91;FIXED JENKINS-2997&#93;</span></p>
<ul class="alternate" type="square">
	<li>don't start two builds of the same project in the same second</li>
	<li>don't schedule two parameterized builds without parameters at the same time
------------------------------------------------------------------------
r15339 | huybrechts | 2009-02-14 04:45:57 -0800 (Sat, 14 Feb 2009) | 1 line
Changed paths:
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java</li>
</ul>


<p><a href="https://issues.jenkins.io/browse/JENKINS-2997" title="multiple parallel parameterized builds seem to happen at once" class="issue-link" data-issue-key="JENKINS-2997"><del>JENKINS-2997</del></a> no need to serialize this...</p>
</pre>
</details>

</ul>
</details>

<p>r15338 | huybrechts | 2009-02-14 04:36:57 -0800 (Sat, 14 Feb 2009) | 3 lines<br/>
Changed paths:<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/ParametersAction.java</p>

<p><span class="error">&#91;FIXED JENKINS-2997&#93;</span></p>
<ul class="alternate" type="square">
	<li>don't start two builds of the same project in the same second</li>
	<li>don't schedule two parameterized builds without parameters at the same time<br/>
------------------------------------------------------------------------<br/>
r15339 | huybrechts | 2009-02-14 04:45:57 -0800 (Sat, 14 Feb 2009) | 1 line<br/>
Changed paths:<br/>
   M /trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java</li>
</ul>


<p><a href="https://issues.jenkins.io/browse/JENKINS-2997" title="multiple parallel parameterized builds seem to happen at once" class="issue-link" data-issue-key="JENKINS-2997"><del>JENKINS-2997</del></a> no need to serialize this...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128674&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128674">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : huybrechts
Path:
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=15339" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=15339</a>
Log:
<a href="https://issues.jenkins.io/browse/JENKINS-2997" title="multiple parallel parameterized builds seem to happen at once" class="issue-link" data-issue-key="JENKINS-2997"><del>JENKINS-2997</del></a> no need to serialize this...</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : huybrechts<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractProject.java<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=15339" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=15339</a><br/>
Log:<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-2997" title="multiple parallel parameterized builds seem to happen at once" class="issue-link" data-issue-key="JENKINS-2997"><del>JENKINS-2997</del></a> no need to serialize this...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128675&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128675">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>one more edge case:
1) start parameterized project with "sleep 120" as build script (any param value)
2) schedule a build with param=B (now waiting in queue)
3) schedule a build with param=A (added to queue since different param value)
4) schedule another build with param=A</p>

<p>Job gets added again, even though a duplicate of job in step 3, because
Queue.add() does getItem(p) which just gets the first Item matching that Task. 
QueueAction.shouldSchedule isn't called for Actions on Item from step 3.</p>

<p>pretty minor case, but thought I'd mention it..</p>
</pre>
</details>

</ul>
</details>

<p>one more edge case:<br/>
1) start parameterized project with "sleep 120" as build script (any param value)<br/>
2) schedule a build with param=B (now waiting in queue)<br/>
3) schedule a build with param=A (added to queue since different param value)<br/>
4) schedule another build with param=A</p>

<p>Job gets added again, even though a duplicate of job in step 3, because<br/>
Queue.add() does getItem(p) which just gets the first Item matching that Task. <br/>
QueueAction.shouldSchedule isn't called for Actions on Item from step 3.</p>

<p>pretty minor case, but thought I'd mention it..</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128676&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128676">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It looks like this bug has resurfaced. I have a parameterized job that gets
executed multiple times in parallel when I queue several instances with
different parameter values.</p>

<p>I can reproduce the bug with a minimal job, using the instructions that
'mindless' left in his previous comment. The issue occurs both with 1.303 and 1.304.</p>

<p>I'm reopening this issue, I'm avail for further information if needed.</p>
</pre>
</details>

</ul>
</details>

<p>It looks like this bug has resurfaced. I have a parameterized job that gets<br/>
executed multiple times in parallel when I queue several instances with<br/>
different parameter values.</p>

<p>I can reproduce the bug with a minimal job, using the instructions that<br/>
'mindless' left in his previous comment. The issue occurs both with 1.303 and 1.304.</p>

<p>I'm reopening this issue, I'm avail for further information if needed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mdonohue:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128677&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128677">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Are they actually executing in parallel?  Sometimes Hudson will allocate an
executor to a job, but it won't start because it is waiting for a resource to
free up.   A script like:
date 
sleep 100
date</p>

<p>Ought to demonstrate whether they have overlapping timestamps when executing.</p>
</pre>
</details>

</ul>
</details>

<p>Are they actually executing in parallel?  Sometimes Hudson will allocate an<br/>
executor to a job, but it won't start because it is waiting for a resource to<br/>
free up.   A script like:<br/>
date <br/>
sleep 100<br/>
date</p>

<p>Ought to demonstrate whether they have overlapping timestamps when executing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128678&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128678">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Here's a more detailed test report with the script you mentioned. I have 3
executors. I started 3 instances of the same 'Test' Job within a few seconds,
with dummy A/B/A param values :</p>

<ul class="alternate" type="square">
	<li>While Test #1 was being executed (and showing in the Executor view), Test #2
and Test #3 remained in the Queue. (OK)</li>
	<li>Then as soon as Test #1 completed, both Test #2 and Test #3 got moved out of
the Queue and appeared in the Executor view with a progress bar showing some
progress for the both of them. (NOK)</li>
	<li>I have pasted the output of Test #1, #2 and #3, which shows that timestamps do
not actually overlap.</li>
</ul>


<p>=&gt; So it's not bad as it first seemed (jobs are not actually executed in
parallel), but I'd expect jobs to stay queued until they are ready to be
executed. This would:</p>
<ul class="alternate" type="square">
	<li>prevent the user from thinking that multiple instances of the same job are
executing simultaneously (the progress bar is particularly misleading).</li>
	<li>free up a valuable Executor slot, letting other jobs (that are ready to start)
execute.</li>
</ul>


<p>Furthermore, I'm pretty sure that I've seen several instances of the more
complex parameterized job that I have overlap, I'll post again if I have more
information that can help track this down.</p>

<p>Thanks for your efforts!</p>

<p>----------------</p>

<p>Test #1
Started by user anonymous
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson1497163093424666761.sh
+ date
Fri May 15 08:43:56 CEST 2009
+ sleep 100
+ date
Fri May 15 08:45:36 CEST 2009
Finished: SUCCESS</p>

<p>Test #2
Started by user anonymous
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson1486533333714628975.sh
+ date
Fri May 15 08:45:36 CEST 2009
+ sleep 100
+ date
Fri May 15 08:47:16 CEST 2009
Finished: SUCCESS</p>

<p>Test #3
Started by user anonymous
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson487910029299787088.sh
+ date
Fri May 15 08:47:16 CEST 2009
+ sleep 100
+ date
Fri May 15 08:48:56 CEST 2009
Finished: SUCCESS</p>
</pre>
</details>

</ul>
</details>

<p>Here's a more detailed test report with the script you mentioned. I have 3<br/>
executors. I started 3 instances of the same 'Test' Job within a few seconds,<br/>
with dummy A/B/A param values :</p>

<ul class="alternate" type="square">
	<li>While Test #1 was being executed (and showing in the Executor view), Test #2<br/>
and Test #3 remained in the Queue. (OK)</li>
	<li>Then as soon as Test #1 completed, both Test #2 and Test #3 got moved out of<br/>
the Queue and appeared in the Executor view with a progress bar showing some<br/>
progress for the both of them. (NOK)</li>
	<li>I have pasted the output of Test #1, #2 and #3, which shows that timestamps do<br/>
not actually overlap.</li>
</ul>


<p>=> So it's not bad as it first seemed (jobs are not actually executed in<br/>
parallel), but I'd expect jobs to stay queued until they are ready to be<br/>
executed. This would:</p>
<ul class="alternate" type="square">
	<li>prevent the user from thinking that multiple instances of the same job are<br/>
executing simultaneously (the progress bar is particularly misleading).</li>
	<li>free up a valuable Executor slot, letting other jobs (that are ready to start)<br/>
execute.</li>
</ul>


<p>Furthermore, I'm pretty sure that I've seen several instances of the more<br/>
complex parameterized job that I have overlap, I'll post again if I have more<br/>
information that can help track this down.</p>

<p>Thanks for your efforts!</p>

<p>----------------</p>

<p>Test #1<br/>
Started by user anonymous<br/>
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson1497163093424666761.sh<br/>
+ date<br/>
Fri May 15 08:43:56 CEST 2009<br/>
+ sleep 100<br/>
+ date<br/>
Fri May 15 08:45:36 CEST 2009<br/>
Finished: SUCCESS</p>

<p>Test #2<br/>
Started by user anonymous<br/>
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson1486533333714628975.sh<br/>
+ date<br/>
Fri May 15 08:45:36 CEST 2009<br/>
+ sleep 100<br/>
+ date<br/>
Fri May 15 08:47:16 CEST 2009<br/>
Finished: SUCCESS</p>

<p>Test #3<br/>
Started by user anonymous<br/>
<span class="error">&#91;workspace&#93;</span> $ /bin/sh -xe /tmp/hudson487910029299787088.sh<br/>
+ date<br/>
Fri May 15 08:47:16 CEST 2009<br/>
+ sleep 100<br/>
+ date<br/>
Fri May 15 08:48:56 CEST 2009<br/>
Finished: SUCCESS</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>huybrechts:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128679&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128679">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>updated description to reflect it only seems dangerous...</p>
</pre>
</details>

</ul>
</details>

<p>updated description to reflect it only seems dangerous...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128680&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128680">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Got another similar problem today with version 1.308: scheduled 7 builds of the
same job (#117 through #123) on a machine with 2 executors. </p>

<p>The first 2 builds went through OK, but then mayhem started: #120 started
<b>before</b> #119 and failed with the following exception:</p>

<p>FATAL: null
java.lang.NullPointerException
	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:192)
	at hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:349)
	at hudson.model.Run.run(Run.java:947)
	at hudson.model.Build.run(Build.java:112)
	at hudson.model.ResourceController.execute(ResourceController.java:93)
	at hudson.model.Executor.run(Executor.java:119)</p>


<p>I'm also attaching a screenshot of the 'build history' panel, which shows
duplicated items, perhaps an indication of what's going on.</p>
</pre>
</details>

</ul>
</details>

<p>Got another similar problem today with version 1.308: scheduled 7 builds of the<br/>
same job (#117 through #123) on a machine with 2 executors. </p>

<p>The first 2 builds went through OK, but then mayhem started: #120 started<br/>
<b>before</b> #119 and failed with the following exception:</p>

<p>FATAL: null<br/>
java.lang.NullPointerException<br/>
	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:192)<br/>
	at hudson.model.AbstractBuild$AbstractRunner.post(AbstractBuild.java:349)<br/>
	at hudson.model.Run.run(Run.java:947)<br/>
	at hudson.model.Build.run(Build.java:112)<br/>
	at hudson.model.ResourceController.execute(ResourceController.java:93)<br/>
	at hudson.model.Executor.run(Executor.java:119)</p>


<p>I'm also attaching a screenshot of the 'build history' panel, which shows<br/>
duplicated items, perhaps an indication of what's going on.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128681&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128681">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created an attachment (id=725)
Corrupted 'build history' panel</p>
</pre>
</details>

</ul>
</details>

<p>Created an attachment (id=725)<br/>
Corrupted 'build history' panel</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128682&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128682">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3154 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3154 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128683&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128683">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : kohsuke
Path:
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractBuild.java
 trunk/hudson/main/core/src/main/java/hudson/model/Run.java
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19570" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19570</a>
Log:
<span class="error">&#91;FIXED JENKINS-2997&#93;</span> NPE fix. This will be in 1.316.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : kohsuke<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/AbstractBuild.java<br/>
 trunk/hudson/main/core/src/main/java/hudson/model/Run.java<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19570" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19570</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-2997&#93;</span> NPE fix. This will be in 1.316.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128684&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128684">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After updating to 1.318, I can confirm the NPE no longer appears in the log.
However, multiple instances of the same job still appear as 'ongoing' (with a
progress bar) in the 'Build Executor Status' and 'Build History' views, as
described in the comment I left on May 15 07:30:02.</p>

<p>I'm reopening this ticket. I'm available for more information if needed.</p>
</pre>
</details>

</ul>
</details>

<p>After updating to 1.318, I can confirm the NPE no longer appears in the log.<br/>
However, multiple instances of the same job still appear as 'ongoing' (with a<br/>
progress bar) in the 'Build Executor Status' and 'Build History' views, as<br/>
described in the comment I left on May 15 07:30:02.</p>

<p>I'm reopening this ticket. I'm available for more information if needed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128685&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128685">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>can you provide steps to see this problem from a clean install of the latest
hudson release?</p>
</pre>
</details>

</ul>
</details>

<p>can you provide steps to see this problem from a clean install of the latest<br/>
hudson release?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mindless.png" title="mindless's avatar" /> mindless:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128686&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128686">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>maxence: please respond to question above, thanks.</p>
</pre>
</details>

</ul>
</details>

<p>maxence: please respond to question above, thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128687&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128687">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry for not replying earlier, the first notification got lost in a sea of
emails...</p>

<p>The problem I described in my last comment on Aug 7 is still accurate and
reproducible in 1.334. Here is a simple way to reproduce the problem:</p>

<ul class="alternate" type="square">
	<li>Make sure you have 2 executors.</li>
	<li>Create a new freestyle job 'Test'.</li>
	<li>Tick 'This build is parameterized' and add a 'Dummy' String parameter.</li>
	<li>Add an 'Execute shell' build step with the following script: 'date &amp;&amp; sleep
100 &amp;&amp; date'.</li>
	<li>Save the job.</li>
	<li>Schedule 5 builds with different parameter values (a/b/c/d/e), quickly enough
so they all get queued.</li>
	<li>Wait for the first build to finish.</li>
	<li>Watch the 'Build history' frame: it will show two builds with a progress bar,
just like on the attached screenshot (following).</li>
</ul>


<p>Hope that helps,
Maxence</p>
</pre>
</details>

</ul>
</details>

<p>Sorry for not replying earlier, the first notification got lost in a sea of<br/>
emails...</p>

<p>The problem I described in my last comment on Aug 7 is still accurate and<br/>
reproducible in 1.334. Here is a simple way to reproduce the problem:</p>

<ul class="alternate" type="square">
	<li>Make sure you have 2 executors.</li>
	<li>Create a new freestyle job 'Test'.</li>
	<li>Tick 'This build is parameterized' and add a 'Dummy' String parameter.</li>
	<li>Add an 'Execute shell' build step with the following script: 'date && sleep<br/>
100 && date'.</li>
	<li>Save the job.</li>
	<li>Schedule 5 builds with different parameter values (a/b/c/d/e), quickly enough<br/>
so they all get queued.</li>
	<li>Wait for the first build to finish.</li>
	<li>Watch the 'Build history' frame: it will show two builds with a progress bar,<br/>
just like on the attached screenshot (following).</li>
</ul>


<p>Hope that helps,<br/>
Maxence</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=128688&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-128688">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created an attachment (id=1018)
Corrupted 'build history' panel (2)</p>
</pre>
</details>

</ul>
</details>

<p>Created an attachment (id=1018)<br/>
Corrupted 'build history' panel (2)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torarvid:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=143039&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-143039">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi. Is there any progress on this issue? If not, is there any workaround?</p>

<p>We have a linux master, and a windows slave (with label "win").</p>

<p>Our .NET apps have matrix build jobs marked to build on label="win" ('cause they need MSBuild). We use gerrit for code review and the gerrit-trigger-plugin on our hudson server. Whenever someone pushes several commits to gerrit, it triggers several hudson builds, most of which get "ABORTED". Very frustrating...</p>

<p>I am willing to do some coding/testing myself, though I have not looked at the hudson sources :-/</p>
</pre>
</details>

</ul>
</details>

<p>Hi. Is there any progress on this issue? If not, is there any workaround?</p>

<p>We have a linux master, and a windows slave (with label "win").</p>

<p>Our .NET apps have matrix build jobs marked to build on label="win" ('cause they need MSBuild). We use gerrit for code review and the gerrit-trigger-plugin on our hudson server. Whenever someone pushes several commits to gerrit, it triggers several hudson builds, most of which get "ABORTED". Very frustrating...</p>

<p>I am willing to do some coding/testing myself, though I have not looked at the hudson sources :-/</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torarvid:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=143519&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-143519">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>... Am I asking in the wrong place? Does nobody but me have this problem?</p>

<p>Just wondering <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>... Am I asking in the wrong place? Does nobody but me have this problem?</p>

<p>Just wondering <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>maxence:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=143744&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-143744">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@toravid, please take a look at issue #5653. Seems like updating to a recent version should fix your problem.</p>

<p>As far as I'm concerned, parameterized builds now work properly (problem fixed).</p>
</pre>
</details>

</ul>
</details>

<p>@​toravid, please take a look at issue #5653. Seems like updating to a recent version should fix your problem.</p>

<p>As far as I'm concerned, parameterized builds now work properly (problem fixed).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>torarvid:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=143836&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-143836">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@maxence, thanks for replying. But alas, my problem still persists after upgrading to 1.386.</p>

<p>So it seems there might be another cause for my problem... Maybe it has something to do with Matrix builds... (We use that to limit build slaves to those running Windows when we build .NET apps).</p>
</pre>
</details>

</ul>
</details>

<p>@​maxence, thanks for replying. But alas, my problem still persists after upgrading to 1.386.</p>

<p>So it seems there might be another cause for my problem... Maybe it has something to do with Matrix builds... (We use that to limit build slaves to those running Windows when we build .NET apps).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=213203&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-213203">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Does anybody experience this issue?</p>
</pre>
</details>

</ul>
</details>

<p>Does anybody experience this issue?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-2997?focusedId=213209&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-213209">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>No new reports in four years, it's safe to assume this is obsolete. Most of the original report (e.g. parallel builds using the same workspace) simply no longer apply anyway.</p>
</pre>
</details>

</ul>
</details>

<p>No new reports in four years, it's safe to assume this is obsolete. Most of the original report (e.g. parallel builds using the same workspace) simply no longer apply anyway.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-3154%22">JENKINS-3154</a>]</i>
